pyproject.toml

```diff
@@ -3,11 +3,7 @@ requires = [
     "pybind11",
     "scikit-build-core[pyproject]>=0.10",
     "swig",
-    "numpy==1.19.2; python_version == '3.8'",
-    "numpy==1.19.5; python_version == '3.9'",
-    "numpy==1.21.6; python_version == '3.10'",
-    "numpy==1.23.5; python_version == '3.11'",
-    "numpy==1.26.2; python_version == '3.12'",
```

Comment:
I do not think this should be replaced without proper testing. The reason is that numpy version for RMS is pinned, and to avoid ABI version issues we should compile on the oldest version of numpy that is supported by a given python version.

Reply:
Agree that we will need to do some thorough testing. A few good indicators so far:

- [numpy claims that ABI is backward compatible from 2 to 1.x.x](https://numpy.org/doc/stable/dev/depending_on_numpy.html#numpy-2-0-specific-advice)
- All of the tests are passing in the RMS 14.2 workflow in this PR which runs against `numpy==1.24.3`